### PR TITLE
ProfitBricks Cloud API v4 support

### PR DESCRIPTION
**Pull request by edevenport**
### What does this PR do?

Updates to support the ProfitBricks Cloud API v4:

* Support for the image aliases.
* Adding new location.

Expanding ProfitBricks cloud provider for a salt-cloud tool to support more commands:

* `list-images`
* `list-locations`
* `create_datacenter`
* Extended configuration with a `delete_volume` option.

Bug fixes:
* Server `vmstate` vs `state` response.

Documentation and test updates.

### Tests written?

Yes

### Commits signed with GPG?

No

### CODE REVIEW

**Comment by rallytime** [comment]
@tonybaloney Can you review this when you get a moment?

**Comment by gtmanfred** [comment]
What are the reasons we are forcing an upgrade to api v4.  Is it still possible to use api v3, or have they shut it down completely?

Can we not manage both side by side based on which version of the library is installed?

**Comment by edevenport** [comment]
@gtmanfred - The ProfitBricks Cloud API v4 implements the much needed `image_alias` feature. Prior to v4, the `image` name or UUID was required which both change monthly for many ProfitBricks supplied images. This means the cloud profile can become invalid each month which can cause confusion for users.

Also, while the v3 API is still available, prior versions are not. It's not known when the v3 API will be deprecated. Forcing the currently supported v4 API will prevent this from becoming an issue.

**Comment by gtmanfred** [comment]
If it is still available i would rather not force it.

This is the same thing we did with rackspace, we supported their v1 api until they totally deprecated it, and ripped it out then.

**Comment by denza** [comment]
@gtmanfred We added compatibility check for the SDKs where the Profitbricks API 3 is used, like you suggested. This way a user will be informed about possible compatibility problem of a SDK he use. Could you please do a review.

**Review by gtmanfred — CHANGES_REQUESTED** [review]
I left a few stylistic comments, other than that, it looks good! Thanks for making this compatible with the older version!

**Comment by denza** [comment]
@gtmanfred  Sorry about that. Now we should be good to go.

**Review by gtmanfred — APPROVED** [review]

**Comment by rallytime** [comment]
Thank you @edevenport! There are a handful of lint violations here: https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/17626/violations/file/salt/cloud/clouds/profitbricks.py/

Can you fix those? This also has a merge conflict as well. Can you resolve that as well?

**Comment by denza** [comment]
@rallytime violations are fixed. Conflict is resolved.

File: doc/topics/cloud/profitbricks.rst [hunk 1]
```diff
@@ -228,5 +259,5 @@ wait_for_timeout
     The timeout to wait in seconds for provisioning resources such as servers.
     The default wait_for_timeout is 15 minutes.
 
-For more information concerning cloud profiles, see :ref:`here
-<salt-cloud-profiles>`.
+For more information concerning cloud profiles, see :doc:`here
```
Comment (rallytime) [inline doc/topics/cloud/profitbricks.rst]: Is there a reason you changed this reference from using `ref` to `doc`? We need to use `ref` here as using `doc` should no longer be done. This is failing the following test: https://jenkins.saltstack.com/job/PR/job/salt-pr-linode-ubuntu14-n/18095/testReport/junit/unit.test_doc/DocTestCase/test_check_for_doc_inline_markup/.

**Comment by denza** [comment]
@rallytime I made a fix you proposed for the test. Could you please review?

**Review by rallytime — APPROVED** [review]

**Pull request merged**